document deprecation of `@uppy/aws-s3-multipart` (#145)
transloadit/uppy.io: docs/uploader/aws-s3-multipart.mdx

import TabItem from '@theme/TabItem';
import UppyCdnExample from '/src/components/UppyCdnExample';

# AWS S3

The `@uppy/aws-s3` plugin can be used to upload files directly to a S3 bucket or
a S3-compatible provider, such as Google Cloud Storage or DigitalOcean Spaces.

> Uploads can be signed using either [Companion][companion docs] or a custom
> signing function.

Author: Maybe we should put the custom function at the end.
```suggestion
Uploads can be signed using either [Companion][companion docs], temporary credentials, or a custom
signing function.
```

```js {10} showLineNumbers
import Uppy from '@uppy/core';
import Dashboard from '@uppy/dashboard';
import AwsS3 from '@uppy/aws-s3';

import '@uppy/core/dist/style.min.css';
import '@uppy/dashboard/dist/style.min.css';

const uppy = new Uppy()
	.use(Dashboard, { inline: true, target: 'body' })
	.use(AwsS3, { companionUrl: 'http://companion.uppy.io' });
```

## API

### Options

#### `shouldUseMultipart(file)`

:::warning

> You must define this option, otherwise you’ll be using the
> [legacy version of this plugin](./aws-s3). If you use the deprecated
> `@uppy/aws-s3-multipart` package, this will be set to `true` by default, which
> is not a recomended setting.

Reviewer: ```suggestion
Not setting this option will result in a suboptimal experience for some of your user’s uploads.
It is best for speed and stability to upload large (100&nbsp;MB+) files with multipart
and small files with regular uploads.
```

Author: We probably want to keep the link to the legacy docs, no?

Reviewer: ```suggestion
Until the next major version, not setting this option uses the [legacy version of this plugin](./aws-s3).
This is a suboptimal experience for some of your user’s uploads.
It is best for speed and stability to upload large (100&nbsp;MB+) files with multipart
and small files with regular uploads.
```

:::

A boolean, or a function that returns a boolean which is called for each file
that is uploaded with the corresponding `UppyFile` instance as argument.

By default, all files are uploaded as multipart. In a future version, all files

> with a `file.size` ≤ 100 MiB will be uploaded in a single chunk, all files

Reviewer: ```suggestion
with a `file.size` ≤ 100&nbsp;MB will be uploaded in a single chunk, all files
```

Author: We'll be using 100MiB, not 100 MB.
```suggestion
with a `file.size` ≤ 100&nbsp;MiB will be uploaded in a single chunk, all files
```

larger than that as multipart.

Here’s how to use it:

```js
uppy.use(AwsS3, {
	shouldUseMultipart(file) {

> 		// Use multipart only for files larger than 100MiB.

Reviewer: ```suggestion
		// Use multipart only for files larger than 100MB
```

Author: `100 * 2 ** 20` is 100 MiB, not 100 MB
